src/core/utils/useGetReservationGroups.ts

```diff
@@ -103,10 +105,18 @@ if (import.meta.vitest) {
 
     it("groups reservations by transaction id", () => {
       const reservations = groupReservations([
-        generateReservation({ transactionId: "t1" }),
-        generateReservation({ transactionId: "t2" }),
-        generateReservation({ transactionId: "t2" }),
-        generateReservation({ transactionId: "t3" })
+        generateReservation({ transactionId: "t1", reservationId: 1 }),
+        generateReservation({
+          reservationId: 2,
+          transactionId: "t2",
+          reservationType: "parallel"
+        }),
+        generateReservation({
+          reservationId: 3,
+          transactionId: "t2",
+          reservationType: "parallel"
+        }),
+        generateReservation({ transactionId: "t3", reservationId: 4 })
```

Comment:
Maybe this is a bit nitpicky, but it would be nice to change reservationId and transactionId like this:

```suggestion
        generateReservation({ transactionId: "t1", reservationId: 1 }),
        generateReservation({
          transactionId: "t2",
          reservationId: 2,
          reservationType: "parallel"
        }),
        generateReservation({
          transactionId: "t2",
          reservationId: 3,
          reservationType: "parallel"
        }),
        generateReservation({ transactionId: "t3", reservationId: 4 })
```